contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -3378,41 +3385,43 @@ grant_perms_to_objects_in_schema(const char *schema_name,
 	while (HeapTupleIsValid(tuple_bbf_schema))
 	{
 		bool isnull;
+		Datum datum;
 		object_name = pstrdup(TextDatumGetCString(heap_getattr(tuple_bbf_schema, Anum_bbf_schema_perms_object_name, dsc, &isnull)));
 		object_type = pstrdup(TextDatumGetCString(heap_getattr(tuple_bbf_schema, Anum_bbf_schema_perms_object_type, dsc, &isnull)));
 		current_permission = DatumGetInt32(heap_getattr(tuple_bbf_schema, Anum_bbf_schema_perms_permission, dsc, &isnull));
-
+		datum = heap_getattr(tuple_bbf_schema, Anum_bbf_schema_perms_function_args, dsc, &isnull);
+		if (!isnull)
+			func_args = pstrdup(TextDatumGetCString(datum));
```

Comment:
you do not need to duplicate the string, copy is already done inside text to cstring